src/Arcus.WebApi.Integration/Hosting/TestHost.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Threading.Tasks;
+using Arcus.Security.Core.Caching;
+using Arcus.WebApi.Jobs.KeyVault;
+using GuardNet;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Xunit.Abstractions;
+
+namespace Arcus.WebApi.Integration.Hosting
+{
+    public class TestHost : WebApplicationFactory<TestStartup>
+    {
+        private readonly TestConfig _config;
+        private readonly ILogger _outputWriter;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TestHost"/> class.
+        /// </summary>
+        public TestHost(TestConfig config, ITestOutputHelper outputWriter)
+        {
+            Guard.NotNull(config, nameof(config));
+            Guard.NotNull(outputWriter, nameof(outputWriter));
+
+            _config = config;
+            _outputWriter = new XunitTestLogger(outputWriter);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="T:Microsoft.Extensions.Hosting.IHostBuilder" /> used to set up <see cref="T:Microsoft.AspNetCore.TestHost.TestServer" />.
+        /// </summary>
+        /// <remarks>
+        /// The default implementation of this method looks for a <c>public static IHostBuilder CreateHostBuilder(string[] args)</c>
+        /// method defined on the entry point of the assembly of <typeparamref name="TEntryPoint" /> and invokes it passing an empty string
+        /// array as arguments.
+        /// </remarks>
+        /// <returns>A <see cref="T:Microsoft.Extensions.Hosting.IHostBuilder" /> instance.</returns>
+        protected override IHostBuilder CreateHostBuilder()
+        {
+            return Host.CreateDefaultBuilder()
+                       .ConfigureWebHost(webHost => webHost.UseStartup<TestStartup>())
+                       .ConfigureAppConfiguration(config => config.AddInMemoryCollection(_config.AsEnumerable()))
+                       .ConfigureServices(services =>
+                       {
+                           var cachedSecretProvider = new Mock<ICachedSecretProvider>();
+                           cachedSecretProvider.Setup(p => p.InvalidateSecretAsync(It.IsAny<string>()))
+                                               .Returns(Task.CompletedTask);
+
+                           services.AddSingleton(serviceProvider => cachedSecretProvider.Object);
+                           services.AddSingleton(_outputWriter);
+                           services.AddServiceBusTopicMessagePump<AutoInvalidateKeyVaultSecretJob>(
+                               subscriptionName: $"Test-{Guid.NewGuid():N}", 
```

Comment:
Do keep in mind that this is related to our Azure resources and might want to be read from the configuration

Reply:
Should I change it now? Meaning place it in the config and read from there?

Reply:
What I would do is:
1. Move subscription name to configuration and rename it to `subscriptionNamePrefix`
2. Append `subscriptionNamePrefix` with the job id since we can have multiple instances

That way, they can influence the names of the subscriptions. If you want to park it in an issue, that's fine for me but I think we'll need it before we ship it.

Reply:
The subscription name is something we use from the `Arcus.Messagin` Azure settings for the background job. Or do you just mean renaming the argument here?

Reply:
That is correct. I'm suggesting to call it `subscriptionNamePrefix`, instead of `subscriptionName`, and append it with the job id.

So `subscriptionNamePrefix: Test-` would create subscription `Test-ABC` for first job.

Reply:
Jep, I fixed that.

Reply:
Did you push your changes because I'm not seeing it? 🤔